VideoLocker/src/main/java/org/edx/mobile/view/CourseDetailFragment.java

```diff
@@ -0,0 +1,263 @@
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import com.bumptech.glide.Glide;
+import com.google.inject.Inject;
+import com.joanzapata.iconify.fonts.FontAwesomeIcons;
+import com.joanzapata.iconify.widget.IconImageView;
+
+import org.apache.http.protocol.HTTP;
+import org.edx.mobile.R;
+import org.edx.mobile.core.IEdxEnvironment;
+import org.edx.mobile.course.CourseDetail;
+import org.edx.mobile.course.GetCourseDetailTask;
+import org.edx.mobile.logger.Logger;
+import org.edx.mobile.util.FileUtil;
+import org.edx.mobile.util.images.CourseCardUtils;
+import org.edx.mobile.util.images.TopAnchorFillWidthTransformation;
+import org.edx.mobile.view.custom.EdxWebView;
+import org.edx.mobile.view.custom.URLInterceptorWebViewClient;
+
+import java.io.IOException;
+
+import roboguice.fragment.RoboFragment;
+import roboguice.inject.InjectExtra;
+
+
+public class CourseDetailFragment extends RoboFragment {
+
+    @Nullable
+    private GetCourseDetailTask getCourseDetailTask;
+
+    private TextView courseTextName;
+    private TextView courseTextDetails;
+    private ImageView headerImageView;
+    private ImageView headerPlayIcon;
+
+    private LinearLayout courseDetailLayout;
+
+    private TextView shortDescription;
+
+    private LinearLayout courseDetailFieldLayout;
+
+    private LinearLayout courseAbout;
+    private EdxWebView courseAboutWebView;
+
+    static public final String COURSE_DETAIL = "course_detail";
+
+    @InjectExtra(COURSE_DETAIL)
+    CourseDetail courseDetail;
+
+
+    protected final Logger logger = new Logger(getClass().getName());
+
+
+    @Inject
+    IEdxEnvironment environment;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        Bundle bundle = this.getArguments();
+        if (bundle != null) {
+            courseDetail = bundle.getParcelable(Router.EXTRA_COURSE_DETAIL);
+        }
+    }
+
+    /**
+     * Sets the view for the Course Card and play button if there is an intro video.
+     */
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        // Course Card View
+        View view;
+        view = inflater.inflate(R.layout.fragment_course_dashboard, container, false);
+        courseTextName = (TextView) view.findViewById(R.id.course_detail_name);
+        courseTextDetails = (TextView) view.findViewById(R.id.course_detail_extras);
+        headerImageView = (ImageView) view.findViewById(R.id.header_image_view);
+        headerPlayIcon = (ImageView) view.findViewById(R.id.header_play_icon);
+        courseDetailLayout = (LinearLayout) view.findViewById(R.id.dashboard_detail);
+
+        headerPlayIcon.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String url = environment.getConfig().getApiHostURL() + courseDetail.media.course_video.uri;
```

Comment:
I don't think we want to prepend the host URL here; I think the course video uri is expected to be a full video URI, no? It might be a http://youtube.com URL for example.